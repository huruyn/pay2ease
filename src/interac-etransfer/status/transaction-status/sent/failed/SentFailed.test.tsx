import { render, screen } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import SentFailed from './SentFailed';

const ReactTestRenderer = require('react-test-renderer');

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe('SentFailed Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <BrowserRouter>
        <SentFailed
          transaction={undefined}
          user={undefined}
          setCurrentStatus={undefined}
        />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render SentFailed', () => {
    render(
      <BrowserRouter>
        <SentFailed
          transaction={undefined}
          user={undefined}
          setCurrentStatus={undefined}
        />
      </BrowserRouter>,
    );

    expect(screen.getByText('From')).toBeInTheDocument();
    expect(screen.getByText('To')).toBeInTheDocument();
    expect(screen.getByText('Transfer Date')).toBeInTheDocument();
    expect(screen.getByText('Transfer Amount')).toBeInTheDocument();
    expect(screen.getByText('Reference Number')).toBeInTheDocument();
  });
});
