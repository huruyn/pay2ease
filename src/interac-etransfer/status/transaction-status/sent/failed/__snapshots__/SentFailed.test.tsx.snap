// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SentFailed Component matches the snapshot 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      From
    </div>
    <div
      className="col-6"
    >
      <strong>
        
      </strong>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      To
    </div>
    <div
      className="col-6"
    >
      <strong>
        
      </strong>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      Transfer Date
    </div>
    <div
      className="col-6"
    >
      
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      Transfer Amount
    </div>
    <div
      className="col-6"
    >
      <strong
        className="amount"
      />
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      <div>
        Reference Number
      </div>
      <div>
        (Keep For Your Records)
      </div>
    </div>
    <div
      className="col-6"
    />
  </div>,
  <div
    className="hstack gap-3"
  >
    <button
      className="zippy-btn zippy-flat d-flex btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Back
    </button>
    <button
      className="zippy-btn d-flex ms-auto btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Send Another Transfer
    </button>
  </div>,
]
`;
